src/main/payment/main/G1HRPS/Payment.java

```diff
@@ -1,164 +1,277 @@
 		payment_id_ = id;
 		guest_id_ = guest_id;
+		room_num_ = room_num;
 		discounts_ = discounts;
 		tax_ = tax;
+		room_charges_ = 0;
+		rs_charges_ = 0;
 		bill_total_ = bill_total;
 		status_ = status;
+	}
+
+	/**
+	 * 
+	 * @return	The payment_id of the payment
+	 */
+
+	public UUID GetPaymentID(){
+
+		return payment_id_;
 
 	}
 
 	/**
+	 * Assigns new guest_id to this payment
+	 * The guest with this guest_id has made this payment
 	 * 
 	 * @param guest_id
 	 */
-	public void SetGuestId(int guest_id) {
-		// TODO - implement Payment.SetGuestId
 
-		this.guest_id_ = guest_id;
+	public void SetGuestID(String guest_id) {
+
+		guest_id_ = guest_id;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetDiscount(float discount) {
-		// TODO - implement Payment.SetDiscount
+	/**
+	 * 
+	 * @return	Guest ID of guest that made payment
+	 */
+
+	public String GetGuestID() {
 
-		this.discounts_ = discount;
+		return guest_id_;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetRoomCharges() {
-		// TODO - implement Payment.SetRoomCharges
+	/**
+	 * Assigns new room_num to this payment
+	 * 
+	 * @param room_num
+	 */
+
+	public void SetRoomNum(int room_num) {
+
+		room_num_ = room_num;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public void SetRoomServices() {
-		// TODO - implement Payment.SetRoomServices
+	/**
+	 * 
+	 * @return	room number of this payment
+	 */
+
+	public int GetRoomNum() {
 
-		throw new UnsupportedOperationException();
+		return room_num_;
+		
 	}
 
 	/**
+	 * Assigns new discount rate to this payment
 	 * 
-	 * @param Tax
+	 * @param discount
 	 */
-	public void SetTax(float Tax) {
-		// TODO - implement Payment.SetTax
 
-		this.tax_ = Tax;
+	public void SetDiscount(float discount) {
+
+		discounts_ = discount;
 
-		throw new UnsupportedOperationException();
 	}
 
 	/**
 	 * 
-	 * @param bill_total
+	 * @return	discount rate of this payment
 	 */
-	public void SetBillTotal(float bill_total) {
-		// TODO - implement Payment.SetBillTotal
 
-		this.bill_total_ = bill_total;
+	public float GetDiscount() {
+
+		return discounts_;
 
-		throw new UnsupportedOperationException();
 	}
 
 	/**
+	 * Assigns room charge of this payment
 	 * 
-	 * @param status
+	 * @param room_charges
 	 */
-	public void SetStatus(PaymentStatus status) {
-		// TODO - implement Payment.SetStatus
 
-		this.status_ = status;
+	public void SetRoomCharges(float room_charges) {
+
+		room_charges_ = room_charges;
+
+	}
+
+	/**
+	 * 
+	 * @return	room charge of this payment
+	 */
+
+	public float GetRoomCharges() {
+		
+		return room_charges_;
 
-		throw new UnsupportedOperationException();
 	}
 
-	public int GetGuestId() {
-		// TODO - implement Payment.GetGuestId
+	/**
+	 * Assign room service charge of this payment 
+	 * 
+	 * @param rs_charges The new room service charge to be assigned
+	 */
 
-		return guest_id_;
+	public void SetRsCharges(float rs_charges) {
```

Comment:
Please rename the method such that it is clear what you're setting. i.e. SetRoomServiceCharges